### DISCUSSION TITLE
Vault import not working

### DISCUSSION THREAD

**Discussion by stanthewizzard** [dani-garcia/vaultwarden#2438] [Q&A]
Versions
Server Installed Ok
1.24.0
Server Latest
1.24.0
Web Installed Ok
2.28.0
Web Latest
2.28.0
Database
MySQL: 10.6.7-MariaDB-1:10.6.7+maria~bullseye-log
Checks
Running within Docker
No
Environment settings overridden
No
Uses a reverse proxy
Yes
IP header Match
Config/Server: X-Real-IP
Internet access Ok
Yes
Internet access via a proxy
No
DNS (github.com) Ok
140.82.121.3
Date & Time (Local)
Server: 2022-04-26 15:48:38 +02:00
Date & Time (UTC) Ok
Server: 2022-04-26 13:48:38 UTC
Browser: 2022-04-26 13:48:39 UTC
Domain configuration Match HTTPS

Hello

When I'm importing json inside the Web GUI
I get this

```
        <html lang="en">
        <head>
            <meta charset="utf-8">
            <title>400 Bad Request</title>
        </head>
        <body align="center">
            <div role="main" align="center">
                <h1>400: Bad Request</h1>
                <p>The request could not be understood by the server due
            to malformed syntax.</p>
                <hr />
            </div>
            <div role="contentinfo" align="center">
                <small>Rocket</small>
            </div>
        </body>
        </html>
```

I'm unable to import
Reverse proxy is nginx
    	client_max_body_size 0;
       large_client_header_buffers 4 16k;

Thanks for help

**Comment by BlackDex** [comment]
It seems to work for me.
I suggest to double check the Proxy configs at the wiki: https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

**Comment by stanthewizzard** [comment]
Nginx (by blackdex) = same issue

when nginx starts I think there is an issue

```
debian--nginx1 vaultwarden[3587]: [2022-04-27 17:01:40.432][request][INFO] GET /api/sync?excludeDomains=true
debian--nginx1 vaultwarden[3587]: [2022-04-27 17:01:40.445][response][INFO] GET /api/sync?<data..> (sync) => 200 OK
debian--nginx1 vaultwarden[3587]: [2022-04-27 17:01:40.574][request][INFO] GET /icons/mondmp1.dmp.gouv.fr/icon.png
debian--nginx1 vaultwarden[3587]: [2022-04-27 17:01:40.576][response][INFO] GET /icons/<domain>/icon.png (icon_internal) => 200 OK
debian--nginx1 vaultwarden[3587]: [2022-04-27 17:02:22.304][request][INFO] PUT /api/ciphers/2d15ce68-860e-47ce-8ab1-62646555af42/delete
debian--nginx1 vaultwarden[3587]: [2022-04-27 17:02:22.309][response][INFO] PUT /api/ciphers/<uuid>/delete (delete_cipher_put) => 200 OK
adebian--nginx1 vaultwarden[3587]: [2022-04-27 17:02:22.325][request][INFO] GET /api/ciphers/2d15ce68-860e-47ce-8ab1-62646555af42
adebian--nginx1 vaultwarden[3587]: [2022-04-27 17:02:22.330][response][INFO] GET /api/ciphers/<uuid> (get_cipher) => 200 OK
debian--nginx1 vaultwarden[3587]: [2022-04-27 17:02:44.691][request][INFO] POST /api/ciphers/import
debian--nginx1 vaultwarden[3587]: [2022-04-27 17:02:44.730][response][INFO] POST /api/ciphers/import (post_ciphers_import) => 400 Bad Request
```

There souldn't be a 400 bad request ?

thanks again

**Reply by BlackDex** [reply]
Also try an other browser or try to disable extensions. Some will interfere by removing headers for example. Or some Anti Virus tools might also be causing issues.

Im using that proxy config (obviously) and all works fine.

**Reply by stanthewizzard** [reply]
saddly same result (brave, firefox, chrome, edge)

db is mariadb (galera cluster) on another debian. Many services are working nicely with it (and yes this homelab lol)

**Reply by stanthewizzard** [reply]
tried also on a mac with safari (totally clean)
doesn't work

**Comment by stanthewizzard** [comment]
sorry to relaunch
could be a the link between nginx post and mariadb ?

**Reply by BlackDex** [reply]
No, because if it reports a `Bad Request`, that means that what is coming in into Vaultwarden isn't correct. So, the JSON is malformed, or some other headers are not correct. etc..

**Reply by stanthewizzard** [reply]
Ok
So I don’t understand

**Reply by BlackDex** [reply]
The database isn't part of the problem. Since the database never was access for this request while this error occurred.

**Comment by olluz** [comment]
same here,
tried in Incognito mode (with all extensions disabled). 
Doesn't work

**Comment by BlackDex** [comment]
From what type are you trying to import. So where does the export come from?

**Reply by olluz** [reply]
Bitwarden exported Data in JSON format. Either as a file or copy/pasteing to the textbox

**Comment by olluz** [comment]
OK, this issue might be related to the file size (at least in my case).
If I remove items from the JSON file so that the size is below 700 kB it starts working...

**Comment by BlackDex** [comment]
Then i suggest to check your revers proxy configuration since it might be that it is limiting the body/post size and that might cut a body in parts which makes the uploaded encrypted json generated by the web-vault invalid.

**Reply by olluz** [reply]
I've followed your Nginx Proxy example. Do I need to configure something else in Nginx for it to work ? 
with `client_max_body_size 128M;` it is not working

**Reply by BlackDex** [reply]
Try to increase that amount. I do not know how large it is, but it could exceed that amount maybe.
You can also set it to `0` which means no checking is done. While i wouldn't recommend that by default, you can check if that solves the issue for now.

Also, if you Press F12 before you post it, you should be able to see the size of the request send to the server in the `network` tab. That might be useful information.
